core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -523,6 +523,11 @@ private <K, V> Cache<K, V> internalGetCache(String cacheName) {
       if (cacheName == null)
          throw new NullPointerException("Null arguments not allowed");
 
+      // Caches retrieved during global registry should only be done post startup so it must not be initializing
+      if (globalComponentRegistry.getStatus() == ComponentStatus.INITIALIZING) {
```

Comment:
Maybe we should use `.allowInvocations()`, in the status? This would allow to retrieve a cache in stopped or failed status.